frameworks/FEDOT/exec_ts.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+from pathlib import Path
+import numpy as np
+
+from fedot.api.main import Fedot
+from fedot.core.repository.tasks import Task, TaskTypesEnum, TsForecastingParams
+from fedot.core.data.data import InputData
+from fedot.core.repository.dataset_types import DataTypesEnum
+
+from frameworks.shared.callee import call_run, result, output_subdir
+from frameworks.shared.utils import Timer, load_timeseries_dataset
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** FEDOT ****\n")
+
+    scoring_metric = get_fedot_metrics(config)
+
+    training_params = {"preset": "best_quality", "n_jobs": config.cores}
+    training_params.update({k: v for k, v in config.framework_params.items() if not k.startswith('_')})
+    n_jobs = training_params["n_jobs"]
+
+    log.info(f"Running FEDOT with a maximum time of {config.max_runtime_seconds}s on {n_jobs} cores, \
+             optimizing {scoring_metric}")
+    runtime_min = config.max_runtime_seconds / 60
+
+    task = Task(
+        TaskTypesEnum.ts_forecasting,
+        TsForecastingParams(forecast_length=dataset.forecast_horizon_in_steps)
+    )
+
+    train_df, test_df = load_timeseries_dataset(dataset)
+    id_column = dataset.id_column
+
+    log.info('Predicting on the test set.')
+    training_duration, predict_duration = 0, 0
+    models_count = 0
+    truth_only = test_df[dataset.target].values
+    predictions = []
+
+
+    for label, train_subdf in train_df.groupby(id_column, sort=False):
+        train_series = train_subdf[dataset.target].to_numpy()
+        train_input = InputData(
+            idx=np.arange(len(train_series)),
+            features=train_series,
+            target=train_series,
+            task=task,
+            data_type=DataTypesEnum.ts
+        )
+
+        test_sub_df = test_df[test_df[id_column] == label].drop(columns=[id_column], axis=1)
+        horizon = len(test_sub_df[dataset.target])
+
+        fedot = Fedot(
+            problem=TaskTypesEnum.ts_forecasting.value,
+            task_params=task.task_params,
```

Comment:
As far as I understand, Fedot currently only supports point forecasting but AMLB may also include probabilistic forecasting tasks (see https://github.com/openml/automlbenchmark/blob/master/amlb/results.py#L767-L792). Probably it would make sense to raise an exception if someone tries to evaluate FEDOT on such a probabilistic forecasting task.

Reply:
is there a way to distinguish a probabilistic forecasting task based on the benchmark run config?

the `get_fedot_metrics` function already emits logs in case of unsupported metrics (like `mql`, `wql`, and `sql`)

Reply:
I think that filtering by the mql, wql, sql metrics is the simplest way to accomplish this.

Another option is to repeat the point forecast for each of the quantile levels and raise a warning.

Reply:
[db10d89](https://github.com/openml/automlbenchmark/pull/611/commits/db10d890c698753cf3edafcf12b81691a1d29309)